gapis/perfetto/cc/processor.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <google/protobuf/text_format.h>
+#include <string.h>
+
+#include "perfetto/trace_processor/raw_query.pb.h"
+#include "perfetto/trace_processor/trace_processor.h"
+
+#include "processor.h"
+
+namespace ptp = perfetto::trace_processor;
+namespace pp = perfetto::protos;
+
+// Callback back into go.
+extern "C" void on_query_complete(int id, void* data, long unsigned int size);
+
+processor new_processor(uint64_t window_size_ns) {
+  ptp::Config config;
+  config.window_size_ns = window_size_ns;
+
+  return ptp::TraceProcessor::CreateInstance(config).release();
+}
+
+bool parse_data(processor processor, const void* data, size_t size) {
+  ptp::TraceProcessor* p = static_cast<ptp::TraceProcessor*>(processor);
+  // TODO: do we need this copy?
+  std::unique_ptr<uint8_t[]> buf(new uint8_t[size]);
```

Comment:
Eww this is an ugly API.
```
virtual bool Parse(std::unique_ptr<uint8_t[]>, size_t) = 0;
```

You will need the copy unless we can allocate the data with `new` but I think this is meant to be exported to go, so the data is coming from "magic place"